code/controllers/subsystem/dbcore.dm

```diff
@@ -62,68 +51,50 @@ SUBSYSTEM_DEF(dbcore)
 	return ..()
 
 /datum/controller/subsystem/dbcore/proc/Connect()
-	if(IsConnected())
-		return TRUE
-
-	if(failed_connection_timeout <= world.time) //it's been more than 5 seconds since we failed to connect, reset the counter
-		failed_connections = 0
-
-	if(failed_connections > FAILED_DB_CONNECTION_CUTOFF)	//If it failed to establish a connection more than 5 times in a row, don't bother attempting to connect for 5 seconds.
-		failed_connection_timeout = world.time + 50
-		return FALSE
-
 	if(!CONFIG_GET(flag/sql_enabled))
 		return FALSE
-
 	var/user = CONFIG_GET(string/feedback_login)
 	var/pass = CONFIG_GET(string/feedback_password)
 	var/db = CONFIG_GET(string/feedback_database)
 	var/address = CONFIG_GET(string/address)
 	var/port = CONFIG_GET(number/port)
+	var/list/connection_result = json_decode(rustg_sql_connect_pool(address, port, user, pass, db, "", "", ""))
+	if(connection_result)
+		var/connection_status = connection_result["status"]
+		world.log << "Database Connection Status: [connection_status]"
+		if(connection_status == "ok")
+			return TRUE
+		else
+			message_admins("WARNING! Database connection failed! Status: [connection_status]")
+			return FALSE
 
-	connection = new /datum/BSQL_Connection(BSQL_CONNECTION_TYPE_MARIADB, CONFIG_GET(number/async_query_timeout), CONFIG_GET(number/blocking_query_timeout), CONFIG_GET(number/bsql_thread_limit))
-	var/error
-	if(QDELETED(connection))
-		connection = null
-		error = last_error
-	else
-		SSdbcore.last_error = null
-		connectOperation = connection.BeginConnect(address, port, user, pass, db)
-		if(SSdbcore.last_error)
-			CRASH(SSdbcore.last_error)
-		UNTIL(connectOperation.IsComplete())
-		error = connectOperation.GetError()
-	. = !error
-	if (!.)
-		last_error = error
-		log_sql("Connect() failed | [error]")
-		++failed_connections
-		QDEL_NULL(connection)
-		QDEL_NULL(connectOperation)
+#warn "REMOVE THIS SHIT BEFORE SHIPPING I SWEAR TO GOD"
+/datum/controller/subsystem/dbcore/proc/TestArbitraryDBCallDebugging(query, parameters = "{}")
+	world.log << "RUNNING QUERY [query]"
+	var/query_results = rustg_sql_query_blocking(query, parameters)
+	world.log << query_results
+	world.log << "QUERY OVER"
 
 /datum/controller/subsystem/dbcore/proc/CheckSchemaVersion()
 	if(CONFIG_GET(flag/sql_enabled))
-		if(Connect())
+		if(rustg_sql_connected())

```

Comment:
this returns json too.
`{"status": "offline"}` or `{"status": "online"}`

worst case scenario, `{"status": "err", "data": "SomethingError: something bad happened!"}`

no wo